### DISCUSSION TITLE
You do not have permissions

### DISCUSSION THREAD

**Discussion by anonymous89002** [smartcontractkit/full-blockchain-solidity-course-py#506] [Q&A] [answered]
How do I update the files permission? When I try installing yarn it gives me an error that I do not have the permissions for the file. I am on Macos, newest version, does anyone know hoe to fix this ?? I have tried changing the permissions already , but when I do it tells me I do not have the necessary permissions to change the permissions. Its annoying please help

**Comment by cromewar** [comment, marked as answer]
hello @anonymous89002 those type of changes requires super user privileges (admin), so in order to execute it you should use `sudo` command, but this is not recommended because npm packages shouldn't have `sudo` permissions. So you would like to install npm and it's packages with NVM or Homebrew.

You can refer to this guide: https://tecadmin.net/install-nvm-macos-with-homebrew/

**Reply by anonymous89002** [reply]
Ok , I have installed this, what command do I use now to have the super user privalages and to have access to these files? Sorry I am new. Thanks in advance

**Reply by cromewar** [reply]
just type `sudo` before any command, it will ask your password. But again is not recommended

**Reply by anonymous89002** [reply]
thanks, it worked

**Reply by cromewar** [reply]
I'm glad it helped.
